addons/common/functions/fnc_exportMissionSQF.sqf

```diff
@@ -0,0 +1,381 @@
+    params ["_object", "_parentVarName"];
+
+    {
+        // Array returned by attachedObjects can contained objNull
+        if (!isNull _x && {isNull isVehicleCargo _x}) then {
+            private _varName = VAR_OBJECT(_x call _fnc_processObject);
+
+            _outputAttach pushBack ["%1 attachTo [%2, %3];", _varName, _parentVarName, _object worldToModel ASLtoAGL getPosASL _x];
+            _outputAttach pushBack ["%1 setVectorDirAndUp %2;", _varName, [vectorDir _x, vectorUp _x]];
+            _outputAttach pushBack "";
+        };
+    } forEach attachedObjects _object;
+};
+
+private _fnc_processUnit = {
+    params ["_unit"];
+
+    private _index = _processedObjects find _unit;
+
+    if (_index == -1) then {
+        _index = _processedObjects pushBack _unit;
+
+        private _groupVarName = VAR_GROUP(_unit call _fnc_processGroup);
+        private _varName = VAR_OBJECT(_index);
+
+        _outputObjects pushBack ["%1 = %2 createUnit [%3, %4, [], 0, ""CAN_COLLIDE""];", _varName, _groupVarName, str typeOf _unit, getPosATL _unit];
+        _outputObjects pushBack ["%1 setDir %2;", _varName, getDir _unit];
+        _outputObjects pushBack ["%1 setRank %2;", _varName, str rank _unit];
+        _outputObjects pushBack ["%1 setSkill %2;", _varName, skill _unit];
+        _outputObjects pushBack ["%1 setUnitPos %2;", _varName, str unitPos _unit];
+
+        if (getForcedFlagTexture _unit != "") then {
+            _outputObjects pushBack ["%1 forceFlagTexture %2;", _varName, str getForcedFlagTexture _unit];
+        };
+
+        if (leader _unit == _unit) then {
+            _outputObjects pushBack ["%1 selectLeader %2;", _groupVarName, _varName];
+        };
+
+        [_unit, _varName] call _fnc_processInventory;
+        [_unit, _varName] call _fnc_processAttachedObjects;
+
+        _outputObjects pushBack "";
+    };
+
+    _index
+};
+
+private _fnc_processVehicle = {
+    params ["_vehicle"];
+
+    private _index = _processedObjects find _vehicle;
+
+    if (_index == -1) then {
+        _index = _processedObjects pushBack _vehicle;
+
+        private _varName = VAR_OBJECT(_index);
+
+        private _placement = if (_vehicle isKindOf "Air" && {isEngineOn _vehicle} && {getPos _vehicle select 2 > 5}) then {
+            "FLY"
+        } else {
+            "CAN_COLLIDE"
+        };
+
+        _outputObjects pushBack ["%1 = createVehicle [%2, [0, 0, 0], [], 0, %3];", _varName, str typeOf _vehicle, str _placement];
+        _outputObjects pushBack ["%1 setVectorDirAndUp %2;", _varName, [vectorDir _vehicle, vectorUp _vehicle]];
+        _outputObjects pushBack ["%1 setPosASL %2;", _varName, getPosASL _vehicle];
+        _outputObjects pushBack ["%1 setDamage %2;", _varName, damage _vehicle];
+        _outputObjects pushBack ["%1 setFuel %2;", _varName, fuel _vehicle];
+
+        _outputObjects pushBack ["{%1 setHitIndex [_forEachIndex, _x, false]} forEach %2;", _varName, getAllHitPointsDamage _vehicle select 2];
```

Comment:
Zero Divisor error on the `select 2`. Happens every time, even on an empty mission. Haven't been able to figure out what exactly causes it.

allMissionObjects "All": 
```
[L Modules:1, Agent 0x7a22c080, Agent 0x348b8180, Agent 0x7a2ac100, L Zeus:1, henk, 2899531: cl_feathers2.p3d, 2899517: cl_grassmulti.p3d, 2899534: cl_feathers2.p3d, 2899526: cl_feathers2.p3d, Agent 0x79564100, Agent 0x69aee140]
```

Reply:
Results from `getAllHitPointsDamage` returning `[]` for some objects.